app/src/main/java/com/github/orkest/View/profile/ProfileTopInterface.kt

```diff
@@ -72,19 +77,51 @@ fun ProfileTopInterface(viewModel: ProfileViewModel) {
 
         Spacer(modifier = Modifier.height(separator))
 
-        Row(){
-            EditButton {
-                val intent = Intent(context, EditProfileActivity::class.java)
-                context.startActivity(intent)
+
+        Row{
+            if(viewModel.username.value == Constants.currentLoggedUser) {
+                EditButton {
+                    val intent = Intent(context, EditProfileActivity::class.java)
+                    context.startActivity(intent)
+                }
+            } else {
+                FollowButton(viewModel, viewModel.isUserFollowed.observeAsState().value)
             }
         }
     }
 }
 
+@Composable
+fun FollowButton(viewModel: ProfileViewModel, isUserFollowed: Boolean?){
+    if(isUserFollowed == null){ Text(text="") } //Empty body, here waiting for the future that fetch isUserFollowed to complete
+    else {
+        Button(
+            onClick = {
+                if (isUserFollowed) { viewModel.unfollow().whenComplete { _, _ -> viewModel.isUserFollowed.value = false } }
+                else { viewModel.follow().whenComplete { _, _ -> viewModel.isUserFollowed.value = true } }
+            },
+            shape = RoundedCornerShape(20.dp),
+            colors = ButtonDefaults.buttonColors(
+                backgroundColor = if (isUserFollowed) Color.White else followColor,
+                contentColor = if (isUserFollowed) followColor else Color.White
+            ),
+            border = BorderStroke(2.dp, followColor),
+            modifier = buttonSettings
+        ) {
+            Text(
+                text = if (isUserFollowed) "Unfollow" else "Follow",
+                fontSize = smallFontSize
+            )
+        }
+    }
+}
+
+
+

```

Comment:
Overall I beleive this is good,  I aslo like the UI experience. Maybe a little more comments could be usefull

Reply:
I will do so, thank you!